src/__init__.py

```diff
@@ -5096,6 +5096,133 @@ def resolve_link(self, uri=None, chapters=0):
         pno = mupdf.fz_page_number_from_location(this_doc, loc)
         return pno, xp, yp
 
+
+    def resolve_names(self):
+        """Convert the PDF's destination names into a Python dict.
+
+        This function must be used with PyMuPDF's new, "rebased" architecture -
+        see the above import statement.
+
+        The only parameter is the fitz.Document.
+        All names found in the catalog under keys "/Dests" and "/Names/Dests" are
+        being included.
+
+        Returns:
+            A dcitionary with the following layout:
+            - key: (str) the name
+            - value: (dict) with the following layout:
+                * "page":  target page number (0-based). If no page number found -1.
+                * "to": (x, y) target point on page - currently in PDF coordinates,
+                        i.e. point (0,0) is the bottom-left of the page.
+                * "zoom": (float) the zoom factor
+                * "dest": (str) only occurs if the target location on the page has
+                        not been provided as "/XYZ" or if no page number was found.
+            Examples:
+            {'__bookmark_1': {'page': 0, 'to': (0.0, 541.0), 'zoom': 0.0},
+            '__bookmark_2': {'page': 0, 'to': (0.0, 481.45), 'zoom': 0.0}}
+
+            or
+
+            '21154a7c20684ceb91f9c9adc3b677c40': {'page': -1, 'dest': '/XYZ 15.75 1486 0'}, ...
+        """
+        # this is a backward listing of page xref to page number
+        page_xrefs = {self.page_xref(i): i for i in range(self.page_count)}
+
+        def obj_string(obj):
+            """Return string version of a PDF object definition."""
+            buffer = fitz.mupdf.fz_new_buffer(512)
+            output = fitz.mupdf.ll_fz_new_output_with_buffer(buffer.m_internal)
+            fitz.mupdf.ll_pdf_print_obj(output, obj.m_internal, 1, 0)
+            printed = fitz.JM_UnicodeFromBuffer(buffer)
+            fitz.mupdf.ll_fz_drop_output(output)
+            fitz.mupdf.ll_fz_drop_buffer(buffer.m_internal)
```

Comment:
Looks like this code i using the low-level `ll_*` interface because there is no `mupdf.fz_new_output_with_buffer()` - see the comment in mupdf/platform/c++/include/mupdf/classes2.h:
```
	/* Class-aware wrapper for `fz_new_output_with_buffer()`
	is not available because returned wrapper class for `fz_output`
	is non-copyable. */
```
However one can instead construct a `mupdf.FzOutput` directly with `output = fitz.mupdf.FzOutput(buffer)`, and then use `fitz.mupdf.pdf_print_obj()` etc, and not have to worry about calling `fitz.mupdf.ll_fz_drop_output()`.

Reply:
Ok, cool so far. But how would I read the content of that FzOutput into a Python string (using UTF8)?